### ISSUE TITLE
Git.Clone.apply is not a function

### ISSUE THREAD

**Issue by marthinus-engelbrecht** [nodegit/nodegit#1139]
Hi please see the below. 

```javascript 

let Git = require('nodegit');

console.log("Git.Clone type: ", typeof Git.Clone); //function
console.log("Git.Clone apply: ", Git.Clone.apply); //undefined

let myFunction = function() {

};

console.log("myFunction type: ", typeof myFunction); //function
console.log("myFunction apply: ", myFunction.apply); //[Function: apply]

```

It seems that somehow Function.prototype.apply gets ripped off the Git.Clone function. This prevents we from using [sinon](http://sinonjs.org/docs/) to create a spy of it.

**Comment by rcjsuen** [comment]
Interesting find. I did some digging and found a few other cases where the `apply()` function was missing. 
```
Cherrypick
Merge
Reset
Revert
```
The common theme I've found between all of these classes are that only have class methods and no instance methods. However, `Ignore` also fulfills this criteria but does not have this problem so something else is different about these particular classes.

Note that if you look at the bottom of the generated `nodegit.js`, you'll find this piece of code for the five offending classes only. I don't see `Ignore` in this list.
```
// Inherit directly from the original Clone object.
_Clone.clone.__proto__ = _Clone;

// Ensure we're using the correct prototype.
_Clone.clone.prototype = _Clone.prototype;

// Assign the function as the root
rawApi.Clone = _Clone.clone;
```
I imagine this is the cause?

**Comment by maxkorp** [comment]
Yes, we do that so you can call `Git.Clone` instead of `Git.Clone.clone` (and so on). 

Don't really want to remove that, but open to figuring out how we work around that. 
Thanks for posting example code, I have a few minutes, lemme try playing around with it
